### ISSUE TITLE
Flex Object directory.config values only accessible if corresponding .yaml file exists

### ISSUE THREAD

**Issue by phmg701** [getgrav/grav#3277]
Hello,

I noticed that options defined in Flex Object blueprint such as:

```
blueprints:
  # Blueprint for configure view.
  configure:
    # We are inside TABS field.
    fields:
      # Add our own tab
      preferences:
        type: tab
        title: Preferences
        fields:
          collection.prefs.test:
            type: number
            size: small
            default: 100
            label: Test (number/int)
            help: Help Text
            validate:
              type: int
              min: 80
              max: 800
```

are only saved in corresponding .yaml file (under `user/config/flex/`) if one of them is modified and saved.

Only then, `directory.config.collection.prefs.test` becomes available in the `directory` object.

I think such fields should be always available, in order to be used in Twig even if Flex Object configuration remains untouched. Usually template logic is based on defaults, but as of now, they would not be accessible.

I found my way by trial and error (and `dump()` ), as Learn just gives a hint for the topic, so there might be another path to access such fields, of which I am unaware.

Thank you!

**Comment by mahagr** [comment]
This is how the configuration works everywhere in Grav. Usually both Grav and plugins/themes define their default values inside a default configuration file. 

You can define the default values inside the blueprint using the `config` section.

**Comment by phmg701** [comment]
Sorry, I didn't make sense, I will try again: I expect Grav not to load data if undefined/YAML file not found, that's a really nice feature indeed, loading just what exists.

What I meant is that from what I get from [Learn (last paragraph)](https://learn.getgrav.org/17/advanced/flex/custom-types/blueprint), custom configuration happens here.
Learn only explains `config -> data`, `config -> site` and `config -> admin`.
How do we use `config -> blueprint` (?) section?

Again, thank you for your time!

**Comment by mahagr** [comment]
I did understand your question. The answer to why it's not doing that is really the speed, it takes some time to get defaults through blueprints which is why it's only being used during editing.

The configuration is really what you see, meaning that `collection.prefs.test` is basically:
```yaml
config:
  collection:
    perfs:
      test: value
```

**Comment by phmg701** [comment]
Thank you, everything clear now!

I did some testing and data from blueprint's `config` is accessible from `directory.config.admin.collection.prefs`, whilst data from YAML file is accessible from `directory.config.collection.prefs`

As far as I understand, no logic (like YAML value overwriting blueprint's value) is available through Object itself, so it has to be implemented in Twig by checking value/field existence, for example.

Thanks again!

**Comment by mahagr** [comment]
Well, the object can override any value by its own logic (yes, you can extend the classes and add any behavior you want to). You can just get the default in twig by `{{ object.perf ?? directory.config.object.prefs }}`. Or you can make objects to get defaults from the directory (just put the values under `directory.config.object` or somewhere you can easily access/merge them. Objects and collections already use `directory.config.cache` properties to customize caching.

**Comment by phmg701** [comment]
Another small step forward for me, thanks!

`{% set prefs = (directory.config.collection.prefs ?? directory.config.admin.collection.prefs) | merge(header.prefs) %}`

Here's how I dealt with priorities and it works!

The only drawback I found is that Twig `merge` isn't recursive, so multidimensional arrays cannot be merged as in plain 
PHP which has separate functions for recursive and non-recursive array merge.

Nothing special, I just used prefixes instead of nesting variables.

Thank you!
